platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/manager/ThreadManager.java

```diff
@@ -16,38 +16,187 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
 
 /**
  * Responsible for managing threading resources.
  */
-public interface ThreadManager extends Lifecycle {
+public interface ThreadManager {
 
     /**
-     * Create a new thread. Thread is not automatically started.
+     * Create a new cached thread pool. If this thread manager has not yet been started, work submitted the executor
+     * service will be not be handled until after the thread manager has been started (with the exception of the
+     * invokeAny() and invokeAll() methods, which will throw if called prior to the thread manager being started).
      *
-     * @param threadGroup
-     * 		the thread group into which the thread is placed
-     * @param runnable
-     * 		the runnable that will be executed on the thread
-     * @return a new Thread
-     * @throws com.swirlds.common.utility.LifecycleException
-     * 		if called before the thread manager has been started
+     * @param name the name of the thread pool
+     * @return a new cached thread pool
      */
-    Thread createThread(ThreadGroup threadGroup, Runnable runnable);
+    @NonNull
+    ExecutorService createCachedThreadPool(@NonNull final String name);
 
     /**
-     * Create a new thread factory. Thread factory will throw
-     * {@link com.swirlds.common.exceptions.MutabilityException MutabilityException} if it is used
-     * to create a thread before the thread manager is started.
+     * Create a new cached thread pool. If this thread manager has not yet been started, work submitted the executor
+     * service will be not be handled until after the thread manager has been started (with the exception of the
+     * invokeAny() and invokeAll() methods, which will throw if called prior to the thread manager being started).
      *
-     * @param component
-     * 		the component name
-     * @param threadName
-     * 		the thread name (each thread created by the factory will also have a unique number
-     * 		appended to the thread name)
-     * @return a thread factory
+     * @param name                     the name of the thread pool
+     * @param uncaughtExceptionHandler handles uncaught exceptions
+     * @return a new cached thread pool
      */
-    ThreadFactory createThreadFactory(final String component, final String threadName);
+    @NonNull
+    ExecutorService createCachedThreadPool(
+            @NonNull final String name, @NonNull final UncaughtExceptionHandler uncaughtExceptionHandler);
+
+    /**
+     * Create a new single thread executor. If this thread manager has not yet been started, work submitted the executor
+     * service will be not be handled until after the thread manager has been started (with the exception of the
+     * invokeAny() and invokeAll() methods, which will throw if called prior to the thread manager being started).
+     *
+     * @param name the name of the thread pool
```

Comment:
throughout this file, the javadoc "the name of the thread pool" is reused, even for methods that aren't creating a thread pool